cmd/otelcontribcol/components.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector/config"
+	"github.com/open-telemetry/opentelemetry-collector/defaults"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/oterr"
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/stackdriverexporter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinscribereceiver"
+)
+
+func components() (config.Factories, error) {
+	errs := []error{}
+	factories, err := defaults.Components()
+	if err != nil {
+		return config.Factories{}, err
```

Comment:
Is there an expectation that this must not return nil factories on error?

Reply:
I wouldn't say so. The calling function shouldn't do anything with the value returned but generally we return nil or zero value on error. This function returns value instead of pointer so can't return nil.

Reply:
Ah, I see, I didn't notice it is return by value.